### PR TITLE
Don't attempt to parse the DTD for Plist files

### PR DESCRIPTION
**Pull request by qmfrederik**
The `Plist` class parses Apple Property List files. These files come with a DTD declaration.

[DTD processing is now prohibited by default in .NET 4.0 and above](https://msdn.microsoft.com/en-us/magazine/ee335713.aspx), but is still enabled (requested) unless you explicitly disable it. This would result in an exception when parsing property list files (for example, when opening DMG images).

This PR disables DTD processing alltogether and adds a unit test for the property list parsing code.

### CODE REVIEW

**Comment by qmfrederik** [comment]
@LordMike Can you take a look at this? Thanks!

**Comment by zivillian** [comment]
> The default value of ProhibitDtd in XmlReaderSettings is true

so just passing a new instance of `XmlReaderSettings` to `XmlReader.Create` would be more "efficient", but your solution seems to be more specific about the desired settings.

> In .NET Framework version 4.0 [...] The ProhibitDtd property has been deprecated 

If I'm correct, your change would introduce a compiler warning.

> you can set the DtdProcessing property to Ignore, which will [...] simply skip over it and not process it

I guess this is what you intended for .NET > 4.0.

(I haven't tested this and the linked documentation mentions only a beta version, so this might be incorrect.)

**Comment by LordMike** [comment]
Totally forgot about this one.

Will we ever encounter PList files that use custom entities?

Would it be better to let DtdProcessing be an option the user sets, through some settings object?

**Comment by qmfrederik** [comment]
@zivillian I perhaps misunderstood your comment; but this PR is not changing the value of `ProhibitDtd`; but it is setting `DtdProcessing` to `Ignore` on .NET 4 and above.

**Comment by qmfrederik** [comment]
@LordMike

> Will we ever encounter PList files that use custom entities?

I'm fairly sure that we will never encounter a Plist file with custom entities. Plist files can be serialized in various formats, XML being one of them. Plist doesn't have the notion of entities so you will not see any entities in the XML files generated by XML.

> Would it be better to let DtdProcessing be an option the user sets, through some settings object?

Basically, on .NET 4.0 and above (this includes Core), `DtdProcessing` is always prohibited, so it must be set to Ignore, otherwise it won't work. You will never want to do DTD processing on Plist files, so I don't think there is any value to letting the user change that value.

**Comment by qmfrederik** [comment]
@LordMike @zivillian Back from vacation, rebased the PR, the build now passes, let me know if this is good to go.

**Comment by zivillian** [comment]
@qmfrederik It looks like I've misread your code - judging the comments and the changes this should be fine.

**Comment by LordMike** [comment]
@qmfrederik put a note in the ifdef where you set the DtdIgnore property explaining that i MUST be set to Ignore on anything by net20.  Then it should be good to go.

File: Library/DiscUtils.Core/Plist.cs [hunk 1]
```diff
@@ -37,7 +37,16 @@ internal static class Plist
 #if !NETCORE
             xmlDoc.XmlResolver = null;
 #endif
-            xmlDoc.Load(stream);
+
+            XmlReaderSettings settings = new XmlReaderSettings();
+#if !NET20
+            settings.DtdProcessing = DtdProcessing.Ignore;
```
Comment (LordMike) [inline Library/DiscUtils.Core/Plist.cs]: Put a note here, explaining that i MUST be set to Ignore on anything by net20. Then it should be good to go.

**Comment by qmfrederik** [comment]
@LordMike Should be good to go now ;-)

**Pull request merged**